atomicapp/plugin.py

```diff
@@ -166,3 +167,10 @@ def getProvider(self, provider_key):
             if key == provider_key:
                 logger.debug("Found provider %s", provider)
                 return provider
+
+        # importing this on top of file leads to circular import
+        from atomicapp.nulecule.exceptions import NuleculeException
```

Comment:
local imports are kind of hacks, and I prefer to avoid them. Let's not reference ``Nulecule*`` things in ``plugin.py`` at all, because, ``plugin.py`` is a dependency for ``Nulecule*`` and other things. Does it make sense?